src/edge_containers_cli/cmds/k8s_commands.py

```diff
@@ -103,10 +104,22 @@ def template(self, svc_instance: Path, args: str):
         )
         chart.deploy_local(svc_instance)
 
-    def deploy_local(self, svc_instance: Path, yes: bool, args: str):
+    def deploy_local(
+        self,
+        svc_instance: Path,
+        yes: bool,
+        args: str,
+        developer: bool,
+    ):
         service_name = svc_instance.name.lower()
 
         chart = Helm(self.namespace, service_name, args=args)
+
+        if developer:
+            image_spec = get_image(svc_instance)
+            dev_image = re.sub("-runtime:", "-developer:", image_spec)
```

Comment:
I'm a little unhappy that ec has now got some hard coded values specific to how we chose to name our containers.

These names come from ioc-template and up to now there has been no dependency between these two projects.

An alternative would be:
Instead of --developer have a --image option in which you pass the image name you want to use instead of the one in values.yaml.

cons:
- its more typing

pros:
- it's more flexible, you could use it for a quick test of a different image version if you wish
- it will work for non IOCs too

What do you think?

Reply:
One thing to be aware of is the assumption that the image is defined in `shared.ioc-instance.image`. Therefore this will not work for non iocs, thats why I have assumed a name specific approach. 

Im not sure I am adding a new dependancy here as this is also already done here: https://github.com/epics-containers/edge-containers-cli/blob/11eca91a414bef3c08aee967b258e4aca0ddaa80/src/edge_containers_cli/utils.py#L66

I would suggest, if we are happy with the idea of pasting the the name of the image... Perhaps its better to just change the image in your `values.yaml`?

Reply:
> One thing to be aware of is the assumption that the image is defined in `shared.ioc-instance.image`. Therefore this will not work for non iocs, thats why I have assumed a name specific approach.
> 
> Im not sure I am adding a new dependancy here as this is also already done here:
> 
> https://github.com/epics-containers/edge-containers-cli/blob/11eca91a414bef3c08aee967b258e4aca0ddaa80/src/edge_containers_cli/utils.py#L66
> 
> I would suggest, if we are happy with the idea of pasting the the name of the image... Perhaps its better to just manually change the image in your service's `values.yaml` and deploy that?

In the same way, I think a `--sleep` option could also just be replaced with the making the entry point overridable by values.yaml

Reply:
I see what you mean. 

My take on this is that it is useful to have a quick way to do this that you won't forget about and accidentally check in to the repo. I also think that given ec's main focus is IOCs that we should keep this PR as is.

Maybe we should change the option description?
- from
  -  Deploy ioc developer image
- to
  - Deploy developer image (IOCs ONLY)

Or something else that highlights the restriction.
In addition I guess we could check that the requested service IS and IOC somehow.

Reply:
in fact this is easy:
```python
if image_spec == dev_image:
  print("ERROR this function only works for containers with both runtime and developer images")
  typer.exit(1)